src/incubator/expandableOverlay/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import {render} from '@testing-library/react-native';
+import ExpandableOverlay, {ExpandableOverlayProps} from '../index';
+import View from '../../../components/view';
+import Text from '../../../components/text';
+import {ExpandableOverlayDriver} from '../ExpandableOverlay.driver';
+
+// What we want to test:
+/**
+ * 1. Sanity - it renders.
+ * 2. Clicking opens the modal/dialog.
+ */
+
+const testID = 'expandableOverylayTest';
+const helloWorld = 'Hello World';
+
+const TestCase = (props: Omit<ExpandableOverlayProps, 'testID'>) => {
+  return (
+    //@ts-expect-error
+    <ExpandableOverlay migrateDialog {...props} testID={testID}>
+      <View>
+        <Text>{helloWorld}</Text>
+      </View>
+    </ExpandableOverlay>
+  );
+};
+
+const getDriver = <T extends ExpandableOverlayProps>(props: T) => {
+  const renderTree = render(<TestCase {...props}/>);
+  const driver = ExpandableOverlayDriver<T>({renderTree, testID});
+  return {driver, renderTree};
+};
+
+describe('ExpandableOverlay', () => {
+  describe('basic functionality', () => {
+    it('should render', () => {
+      render(<TestCase/>);
+    });
+    it('should open dialog when pressed', () => {
+      const {driver} = getDriver({useDialog: true});
+      expect(driver.getOverlay().exists()).toBeFalsy();
+      expect(driver.getOverlay().getModal().exists()).toBeTruthy();
```

Comment:
Not sure this makes sense (I am not saying it's not working ATM, but does it make sense?).
How can the `Modal` exist if there's no `Overlay`?
I feel like this is implementation based; shouldn't we test for visibility?

Reply:
changed it to test only for visibility.